apps/site/test/site_web/controllers/schedule/line_test.exs

```diff
@@ -446,4 +448,19 @@ defmodule SiteWeb.ScheduleController.LineTest do
       assert html_response(conn, 200) =~ "Forest Hills"
     end
   end
+
+  describe "get_shuttle_data" do
+    test "returns empty data when given a route and a date_time without shuttle service", %{
+      conn: conn
+    } do
+      conn =
+        conn
+        |> Conn.assign(:route, %Route{id: "Red", type: 1})
+        |> Conn.assign(:date_time, ~N[2019-11-25T12:00:00])
+
+      expected = %Site.ShuttleDiversion{shapes: [], stops: []}
+      data = get_shuttle_data(conn)
+      assert data == expected
```

Comment:
I'm not sure this test is valuable to us as-is — we'd expect the data to be empty for any date in the past, since the relevant Alert would no longer exist, and we can't pick a date in the future since the alerts won't exist yet.

I think what we want is for `ShuttleDiversion` itself to have comprehensive tests, which is something I'll be working on soon. If that's the case, we can be confident that the very straightforward `get_shuttle_data` will work as we expect. But if we still wanted to unit-test it, we could mock `ShuttleDiversion` (look up `with_mock` for how we're currently doing this, which is not great, but it's something).

Reply:
Unit tests on `ShuttleDiversion` sounds like a good idea. As for mocking `ShuttleDiversion`, I'm not quite grasping how that would help with this particular test. Let's discuss soon. :)

Reply:
`mock` works like this:

```elixir
import Mock

diversion_fn = fn _, _ -> {:ok, %ShuttleDiversion{...}} end

with_mock ShuttleDiversion, active: diversion_fn do
  # in here, calls to `ShuttleDiversion.active` are instead sent to `diversion_fn`
end
```

This is slightly dangerous because it doesn't verify that `diversion_fn` actually matches the signature of `ShuttleDiversion.active` — if the latter were modified (maybe a third argument is added), the test would continue to pass, but would no longer be indicative of what would happen in a real un-mocked environment.